src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp

```diff
@@ -348,10 +348,14 @@ StorageEmbeddedRocksDB::StorageEmbeddedRocksDB(const StorageID & table_id_,
         const StorageInMemoryMetadata & metadata_,
         bool attach,
         ContextPtr context_,
-        const String & primary_key_)
+        const String & primary_key_,
+        Int32 ttl_,
```

Comment:
How can I set the ttl value as a user? 
Why is it signed?

Reply:
`ttl` is the attribute of the storage(Rocksdb), do we need to support set different `ttl` value for different users?

The `ttl` of `DBWithTTL` is signed, and not specifying/passing or non-positive TTL behaves like TTL = infinity.

Reply:
why impose ttl on the users if they don't need it?

Reply:
sorry for misunderstanding the meaning of `user`, I thought it was the `user` in database :).

It is a question that needs to be discussed. In RocksDB, `DB` and `DBWithTTL` are two different type. There is no option like `ttl` and `read_only` in rocksdb options. So we may need to set like
```
CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
(
    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1],
    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2],
    ...
) ENGINE = EmbeddedRocksDB(ttl, read_only) PRIMARY KEY(primary_key_name)
```
which causes no compatibility with previous versions.

Another way is to add a new engine name `EmbeddedRocksDBWithTTL`, use it like
```
CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
(
    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1],
    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2],
    ...
) ENGINE = EmbeddedRocksDBWithTTL(ttl, read_only) PRIMARY KEY(primary_key_name)
```
But it looks wired to add an engine just for TTL.

Any suggestions?

Reply:
> sorry for misunderstanding the meaning of user, I thought it was the user in database :).

No problem, I see how that mistake would be made, glad we are on the same page now.

I would have it like this:
```
CREATE TABLE [IF NOT EXISTS] [db.]table_name [ON CLUSTER cluster]
(
    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1],
    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2],
    ...
) ENGINE = EmbeddedRocksDB([ttl, read_only]) PRIMARY KEY(primary_key_name)
```
You just check the size of arguments and if none are specified you set to a default which will produce same behavior as older versions.

 > In RocksDB, DB and DBWithTTL are two different type
 
 But DBWithTTL inherits from DB, so you can create unique_ptr<DBWithTTL> if ttl is specified, otherwise unique_ptr<DB> and store both of them inside unique_ptr<DB> variable.
 Did I miss something?

Reply:
Seems good. I will implement this later. Thank you~